src/main/java/crawlercommons/sitemaps/sax/XMLHandler.java

```diff
@@ -72,7 +72,20 @@
     public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
         // flush any unclosed or missing URL element
         if (loc.length() > 0 && ("loc".equals(qName) || "url".equals(qName))) {
-            maybeAddSiteMapUrl();
+            // check whether loc isn't white space only
+            for (int i = 0; i < loc.length(); i++) {
+                if (!Character.isWhitespace(loc.charAt(i))) {
+                    maybeAddSiteMapUrl();
+                    return;
+                }
+            }
+            loc = new StringBuilder();
```

Comment:
we probably want to add it not just when a new URL element is found but also a loc (see line 74 above) so why not just call maybeAddSiteMapUrl?